File: examples/read_multi.js
// example reads multiple adc inputs
// run:
// > node read_multi.js

"use strict";

const PORT = 1; // i2c-1
const RAWDATA = false;

const adc = require("../ads1115");

var mux_arr = [adc.MUX_I0_GND, adc.MUX_I1_GND, adc.MUX_I2_GND, adc.MUX_I3_GND]; 

adc.read_multi(
    PORT,
    adc.ADR_48, // address 0x48
    mux_arr,
    adc.GAIN_4096, // 4096mV
    adc.RATE_128, // 128 SPS
    RAWDATA, // raw data ?
    function(data) {
        if (data === undefined)
            console.log(adc.error_text());
        else
            console.log(data, "mV");
    });

File: cpp/read_multi.cpp
/*
 * example reads single adc input and prints value
 *
 * build:
 * > make
 *
  * run:
 * > ./read_multi
 *
 * read_multi.cpp
 *
 */

#include <stdio.h>

#include "ads1115_lib.h"

#define PORT 1
#define RAWDATA false

uint8_t mux_arr[] = { ads1115::MUX_I0_GND, ads1115::MUX_I1_GND, ads1115::MUX_I2_GND, ads1115::MUX_I3_GND };

int main()
{
    double value;

    for(uint32_t i=0; i < 4; i++)
    {
        bool ret = ads1115::read(
            PORT,
            ads1115::ADR_48,
            mux_arr[i],
            ads1115::GAIN_4096,
            ads1115::RATE_128,
            RAWDATA,
            value);

        if (ret)
            printf("ch: %d %.0fmV\n", i, value);
        else
        {
            printf("%s\n", ads1115::error_text());
            break;
        }
    }

    return 0;
}


